doc/changes/0.4.rst

```diff
@@ -3,3 +3,4 @@
 Version 0.4 (in progress)
 ---------------------------
 - Add support for weights and positive coefficients to :ref:`MCPRegression Estimator <skglm.MCPRegression>` (PR: :gh:`184`)
+- Expose a datafit method called by solvers to get Lipschitz. Now on, ``initialize_*`` is meant to initialize internal attributes of the datafit. (PR: :gh:`192`)
```

Comment:
Something like "Move solver specific computations (eg Lipschitz constants) from Datafit.initialize() to separate Datafit methods (eg get_lipschitz_constant), to ease Solver/Datafit compatibility check" makes it less about the aprticular case of Lipschitz constants and more about the general principle